src/commands/commit.ts

```diff
@@ -61,58 +62,127 @@ ${chalk.grey('——————————————————')}
+          );
+          if (stdout) outro(stdout);
+        }
+      } else {
+        const selectedRemote = (await select({
+          message: 'Choose a remote to push to',
+          options: remotes.map((remote) => ({ value: remote, label: remote }))
+        })) as string;
+
+        if (!isCancel(selectedRemote)) {
+          const pushSpinner = spinner();
+          pushSpinner.start(`Running \`git push ${selectedRemote}\``);
+          const { stdout } = await execa('git', ['push', selectedRemote]);
+          pushSpinner.stop(
+            `${chalk.green(
+              '✔'
+            )} successfully pushed all commits to ${selectedRemote}`
+          );
+
+          if (stdout) outro(stdout);
+        } else outro(`${chalk.gray('✖')} process cancelled`);
+      }
+    } else if (isCommitConfirmedByUser == "edit" && !isCancel(isCommitConfirmedByUser)) {
+      await promptUserEdit(commitText)
+    } else if (isCommitConfirmedByUser == "no" && !isCancel(isCommitConfirmedByUser)) {
+      outro(`👋 exiting`);
+    } 
+  }
+  
+  const promptUserEdit = async(commitText: string) => {
+    let commitLines = commitText.split("\n") 
+    let updatedCommitMessage
+
+    if (commitLines.length > 1 ) {
+      let newCommitLines = []
+
+      for (const line of commitLines) {
+        if (line !== "") {
+          const newLine = await text({
+            message: 'Update line:',
+            initialValue: line,
+            placeholder: "",
+            validate(value) {
+              if (line == commitLines[commitLines.length -1] && newCommitLines.length == 0 && value == "") {
+                return "Message cannot be empty" 
+              } 
+            }
+          })
+          
+          if (typeof newLine == 'symbol') { 
+            process.exit(0) 
+          }
+
+          if (newLine !== undefined) {
+            newCommitLines.push(newLine)
+          }
+        }
       }
+      
+      updatedCommitMessage = newCommitLines.join("\n\n")
+
     } else {
-      const selectedRemote = (await select({
-        message: 'Choose a remote to push to',
-        options: remotes.map((remote) => ({ value: remote, label: remote }))
-      })) as string;
-
-      if (!isCancel(selectedRemote)) {
-        const pushSpinner = spinner();
-        pushSpinner.start(`Running \`git push ${selectedRemote}\``);
-        const { stdout } = await execa('git', ['push', selectedRemote]);
-        pushSpinner.stop(
-          `${chalk.green(
-            '✔'
-          )} successfully pushed all commits to ${selectedRemote}`
-        );
-
-        if (stdout) outro(stdout);
-      } else outro(`${chalk.gray('✖')} process cancelled`);
+      updatedCommitMessage = await text({
+        message: 'Update the commit message:',
+        initialValue: commitLines[0],
+        validate: (value) => value == "" ? "Message cannot be empty" : ""
+      })
+    }
+    
+    if (typeof updatedCommitMessage == 'symbol') { 
+      process.exit(0) 
```

Comment:
I believe clack/prompt/text uses 'symbol' for termination signals. I tried to keep the behavior within this function consistent with the rest of the program (i.e. ctrl + c  looks and exits the same way).

Reply:
it exports `isCancel(updatedCommitMessage)` function, you should use it for consistency

Reply:
ha. Idk how I totally missed isCanceled. Thanks for the heads up.